### ISSUE TITLE
`difficulty` 初始化有bug 

### ISSUE THREAD

**Issue by c7h12** [open-spaced-repetition/py-fsrs#73]
这里限制了difficulty在1-10，但是数字是整数类型，
https://github.com/open-spaced-repetition/py-fsrs/blob/f18100d29c774f981f873361d444f8db12e88f77/src/fsrs/fsrs.py#L666

这里assert要求是float
https://github.com/open-spaced-repetition/py-fsrs/blob/f18100d29c774f981f873361d444f8db12e88f77/src/fsrs/fsrs.py#L390
https://github.com/open-spaced-repetition/py-fsrs/blob/f18100d29c774f981f873361d444f8db12e88f77/src/fsrs/fsrs.py#L400

导致报错
![image](https://github.com/user-attachments/assets/387280db-feea-43b5-a8d1-0494ac98e40a)

**Comment by joshdavham** [comment]
Could you give me steps to reproduce this error?

**Comment by c7h12** [comment]
```py
from fsrs import Scheduler, Card, Rating, ReviewLog,State

scheduler = Scheduler()
card = Card()

history = [(card,None)]

for i in range(10):
    print("-"*32,i)
    lastcard,_ = history[-1]
    print("last:",lastcard.to_dict())
    updated,reviewlog = scheduler.review_card(lastcard,Rating.Hard)
    print("updated:",updated.to_dict(),reviewlog)
    history.append((updated,reviewlog))
```

当连续几次Hard,或者Easy，将触碰到`difficulty`的上下界

![image](https://github.com/user-attachments/assets/0c2f64b8-0a27-47b5-a3c7-6e94e49e25ed)

**Comment by joshdavham** [comment]
Thanks.

What version are you using? I tried running your code snippet above in py-fsrs v4.1.1 and I don't get this error. I think this issue was most likely fixed in [v4.1.0](https://github.com/open-spaced-repetition/py-fsrs/pull/67). Are you using an earlier version?

You can check your version with `pip show fsrs`.

**Comment by c7h12** [comment]
感谢回复

报错的是4.0.0，升级4.1.0，不再报错了。
